static_src/stores/user_store.js

```diff
@@ -109,14 +109,19 @@ export class UserStore extends BaseStore {
         const user = this.get(action.userGuid);
         const addedRole = action.roles;
         if (user) {
-          if (!user.roles) user.roles = {};
-          const updatedRoles = new Set(user.roles[action.entityGuid] || []);
-          updatedRoles.add(addedRole);
-          user.roles[action.entityGuid] = Array.from(updatedRoles);
-
+          if (action.entityType === 'space') {
+            if (!user.space_roles) user.space_roles = {};
```

Comment:
How about breaking these up into a method? Both paths seem fairly similar, and a descriptive method call never hurts readability.